.travis.yml

```diff
@@ -52,6 +52,9 @@ script:
   - bash travis-scripts/install-prestashop
   - if [ $PRESTASHOP_TEST_TYPE = "lint" ]; then
         bash tests/check_file_syntax.sh;
+
+        # Check our current PHP Coding Style rules
+        php ./vendor/bin/php-cs-fixer fix --dry-run --stop-on-violation --show-progress=dots
```

Comment:
This should be run only on the modified files, and show all failures (not just the first one)

Reply:
@eternoendless depends, this my previous comment :)

Reply:
I guess the value is that it makes you learn?

If you decide to keep the `stop-on-violation` rule then it should display a message explaining how to fix the errors so that there's no back-and-forth waiting for the build to pass.

In any case, it will be a lot faster if it only checks modified files.

Reply:
Not really faster, maybe 1 or 2 seconds faster for a CI it's nothing :/

Reply:
Let's wait for travis and find out how long it actually takes!

Reply:
it's not up to me to decide ^^ we still need the so famous-called meeting to decide together. Then, I can apply what has been decided.

Also, we can make this step in another build thanks to the jobs feature of Travis: you can look at something I've done in @prestonBot [repository](https://travis-ci.org/PrestaShop/prestonbot/builds/402049713?utm_source=github_status&utm_medium=notification). This way we could  remove ``stop-on-violation`` and I'm almost sure we can ask PrestonBot to ask people to execute the right command to fix CS only if the "CS" jobs fail.

> In any case, it will be a lot faster if it only checks modified files.

I agree, but discussing with @dunglas (member of Symfony core team), I have the feeling that we should apply on all files in case of PHP-CS-Fixer updates. If the issue is only the performance, an extra (and not required) job will do the job (ahah)

Reply:
Don't forget that our tests in the CI are really really fast when many open source projects have their tests taking more than one hour with fast testing infrastructure :P

Reply:
In API Platform we moved "lint jobs" (PHP CS Fixer and PHPStan) to CircleCI so we can parallelize more: https://github.com/api-platform/core/pull/2004

Also, there is also a free service for OpenSource projects that is blazing fast: https://styleci.io/
It has been created by Laravel maintainers IIRC.

Reply:
Thanks for your feedbacks @dunglas :-)